open-api/rest-catalog-open-api.yaml

```
@@ -537,6 +537,113 @@ paths:
         5XX:
           $ref: '#/components/responses/ServerErrorResponse'
 
+  /v1/{prefix}/namespaces/{namespace}/tables/{table}/preplan:
+    parameters:
+      - $ref: '#/components/parameters/prefix'
+      - $ref: '#/components/parameters/namespace'
+      - $ref: '#/components/parameters/table'
+    post:
+      tags:
+        - Catalog API
+      summary: Prepare a list of plan tasks that can be used later for table scan planning
+      description:
+        Prepare a list of plan tasks that can be used later for table scan planning.
+        
+        Each plan task in the response of this API can be used as the `plan-task` in the `PlanTable` 
+        API request to perform scan planning against a subset of the table files.
+        
+        This can be used to parallelize and distribute table scan planning.
+      operationId: PreplanTable
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PreplanTableRequest'
+      responses:
+        200:
+          $ref: '#/components/responses/PreplanTableResponse'
+        400:
+          $ref: '#/components/responses/BadRequestErrorResponse'
+        401:
+          $ref: '#/components/responses/UnauthorizedResponse'
+        403:
+          $ref: '#/components/responses/ForbiddenResponse'
+        404:
+          description:
+            Not Found
+            - NoSuchTableException, the table does not exist
+            - NoSuchNamespaceException, the namespace does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+              examples:
+                TableDoesNotExist:
+                  $ref: '#/components/examples/NoSuchTableError'
+                NamespaceDoesNotExist:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+        419:
+          $ref: '#/components/responses/AuthenticationTimeoutResponse'
+        503:
+          $ref: '#/components/responses/ServiceUnavailableResponse'
+        5XX:
+          $ref: '#/components/responses/ServerErrorResponse'
+
+  /v1/{prefix}/namespaces/{namespace}/tables/{table}/plan:
+    parameters:
+      - $ref: '#/components/parameters/prefix'
+      - $ref: '#/components/parameters/namespace'
+      - $ref: '#/components/parameters/table'
+    post:
+      tags:
+        - Catalog API
+      summary: Perform scan planning against a table
+      operationId: PlanTable
+      description:
+        Perform scan planning against a table based on the user's scan requirements including selected columns and filters.
```

```suggestion
        Perform scan planning against a table based scan requirements including selected columns and filters.
```
I would probably just omit `the user` here

---

How about we simplify it further to `Perform scan planning against a table based on the selected columns and filters`?

---

I feel like this does not read super well either, as we use the word "scan" twice in this statement.

---

Just having `Perform scan planning against a table based on requirements including selected columns and filters.` would probably also work here